### PR TITLE
Add new constructor and get() method to Some<>.

### PR DESCRIPTION
**Pull request by geektoni**
@lisitsyn 

Changes:
* Fix the SGObject equals() (in view of SG_REF refactor);
* get() method permits to extract the raw pointer from Some<> (in view of SG_REF refactor);

Other changes:
* Add .idea (CLion IDE directory) to .gitignore;

### CODE REVIEW

**Comment by lisitsyn** [comment]
:+1:

File: src/shogun/base/some.h [hunk 1]
```diff
@@ -36,13 +39,19 @@ namespace shogun
 		 */
 		T* operator->() const;
 
+		/**
+		 * Get the raw pointer
+		 * @return raw pointer
```
Comment (lisitsyn) [inline src/shogun/base/some.h]: It would make sense to say if it references the pointer

**Comment by geektoni** [comment]
Are CIs failing at random?

**Comment by vigsterkr** [comment]
mmm travis tend to do these things lately:
```
The command "docker exec -t devenv /bin/sh -c "cd /opt/shogun/build; ctest --output-on-failure -j 2"" exited with 0.

Done. Your build exited with 1.
```
which seems to be due to
```
clang: error: unable to execute command: Killed

clang: error: clang frontend command failed due to signal (use -v to see invocation)

Ubuntu clang version 3.4-1ubuntu3 (tags/RELEASE_34/final) (based on LLVM 3.4)

Target: x86_64-pc-linux-gnu

Thread model: posix

clang: note: diagnostic msg: PLEASE submit a bug report to http://bugs.debian.org/ and include the crash backtrace, preprocessed source, and associated run script.

clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.

make[2]: *** [tests/unit/CMakeFiles/shogun-unit-test.dir/clone_unittest.cc.o] Error 254

make[1]: *** [tests/unit/CMakeFiles/shogun-unit-test.dir/all] Error 2

make[1]: *** Waiting for unfinished jobs....

[ 87%] Building CXX object tests/unit/CMakeFiles/shogun-serialization-unit-test.dir/base/main_unittest.cc.o

[ 87%] Building CXX object tests/unit/CMakeFiles/shogun-serialization-unit-test.dir/utils/Utils.cpp.o

Linking CXX executable ../../bin/shogun-serialization-unit-test

Discovering Tests in shogun-serialization-unit-test

[ 87%] Built target shogun-serialization-unit-test

make: *** [all] Error 2

```
i reckon we are running out of mem?

**Comment by vigsterkr** [comment]
@geektoni i've just restarted the windows based one

**Comment by geektoni** [comment]
@vigsterkr AppVeyor is now green :tada:

**Pull request merged**